### PR TITLE
Add siwe

### PR DESCRIPTION
**Pull request by Nazeeh21**

### CODE REVIEW

File: pages/api/hello.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
-import type { NextApiRequest, NextApiResponse } from 'next'
+import type { NextApiRequest, NextApiResponse } from 'next';
```
Comment (sarthakkundra) [inline pages/api/hello.ts:2]: let's remove this file?

File: pages/index.tsx [hunk 2]
```diff
@@ -1,16 +1,69 @@
 import type { NextPage } from 'next';
 import { ConnectButton } from '@rainbow-me/rainbowkit';
+import { SiweMessage } from 'siwe';
+import { useAccount, useSigner } from 'wagmi';
 
 const Home: NextPage = () => {
+  const BACKEND_ADDR = 'http://localhost:8080/';
```
Comment (sarthakkundra) [inline pages/index.tsx:7]: make a separate file, `constants.ts` and add this there

File: pages/index.tsx [hunk 3]
```diff
@@ -1,16 +1,69 @@
...
+  const { address } = useAccount();
+  const { data: signer } = useSigner();
+
+  async function createSiweMessage(address: string, statement: string) {
+    const domain = window.location.host;
```
Comment (sarthakkundra) [inline pages/index.tsx:12]: Let's use conditional chaining and take advantage of TS. `const domain = window?.location?.host`

File: pages/index.tsx [hunk 4]
```diff
@@ -1,16 +1,69 @@
...
+    });
+    return message.prepareMessage();
+  }
+
+  async function signInWithEthereum() {
+    const message = await createSiweMessage(
+      address!,
+      'Sign in with Ethereum to the app.',
+    );
+    console.log(message);
```
Comment (sarthakkundra) [inline pages/index.tsx:34]: remove log

File: pages/index.tsx [hunk 5]
```diff
@@ -1,16 +1,69 @@
...
+    const signature = await signer!.signMessage(message);
+    console.log(signature);
```
Comment (sarthakkundra) [inline pages/index.tsx:36]: remove log

File: pages/index.tsx [hunk 6]
```diff
@@ -1,16 +1,69 @@
...
+  async function sendForVerification(message: string, signature: string) {
+    const res = await fetch(`${BACKEND_ADDR}auth/verify`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ message, signature }),
+      credentials: 'include',
+    });
+    console.log(await res.text());
```
Comment (sarthakkundra) [inline pages/index.tsx:49]: remove log

**Review by sarthakkundra — CHANGES_REQUESTED** [review]

File: pages/api/hello.ts (hunk 1 above)
Comment (Nazeeh21) [reply pages/api/hello.ts:2]: I think we will require it in the future. But we can always add it again whenever we require it, so deleting now

**Pull request merged**